src/layout.js

```diff
@@ -39,7 +39,9 @@ export function calculateX(nodes, data) {
       }
     });
 
-  return x;
+  return [...nodes.values()]
+    .map((node) => node.x)
+    .reduce((n, acc) => Math.max(n, acc), 0);
```

Comment:
This works, but lets create a little less garbage:

```suggestion
  return nodes.values()
    .reduce((max, node) => Math.max(max, node.x), 0);
```